src/fungible_token.ts

```diff
@@ -0,0 +1,103 @@
+import { DUST_AMOUNT, SignerProvider, TransactionBuilder } from '@alephium/web3'
+import { Network } from './network'
+import { TokenFaucet, Withdraw } from '../artifacts/ts'
+import { balanceOf, getNodeUrl, waitTxConfirmed } from './utils'
+import { MaxU256 } from './consts'
+
+export class FungibleToken {
+  network: Network
+  nodeUrl: string
+  builder: TransactionBuilder
+
+  constructor(network: Network) {
+    this.network = network
+    this.nodeUrl = getNodeUrl(network)
+    this.builder = TransactionBuilder.from(this.nodeUrl)
+  }
+
+  static async deploy(
+    signer: SignerProvider,
+    supply: bigint = 0n,
+    name: string = '',
+    symbol: string = '',
+    decimals: bigint = 0n
+  ): Promise<string> {
+    const deployResult = await waitTxConfirmed(
+      TokenFaucet.deploy(signer, {
+        initialFields: {
+          name: Buffer.from(name, 'utf8').toString('hex'),
+          symbol: Buffer.from(symbol, 'utf8').toString('hex'),
+          decimals,
+          supply,
```

Comment:
If we are implementing this design the totalSupply should be also set to max u256